bindings/rust/src/lib.rs

```diff
@@ -413,30 +435,34 @@ impl Instance {
         name: &str,
         args: &[TypedValue],
         depth: i32,
-    ) -> Result<TypedExecutionResult, ()> {
+    ) -> Result<TypedExecutionResult, Error> {
         let func_idx = self.find_exported_function_index(&name);
         if func_idx.is_none() {
-            return Err(());
+            return Err("function not found".into());
         }
         let func_idx = func_idx.unwrap();
 
         let func_type = unsafe { self.get_function_type(func_idx) };
         if func_type.inputs_size != args.len() {
-            return Err(());
+            return Err("argument count mismatch".into());
         }
 
         // Validate input types.
         let supplied_types: Vec<sys::FizzyValueType> = args.iter().map(|v| v.get_type()).collect();
         let expected_types =
             unsafe { std::slice::from_raw_parts(func_type.inputs, func_type.inputs_size) };
         if expected_types != supplied_types {
-            return Err(());
+            return Err("argument type mistmatch".into());
         }
 
         // Translate to untyped raw values.
         let args: Vec<Value> = args.iter().map(|v| v.into()).collect();
 
         let ret = unsafe { self.unsafe_execute(func_idx, &args, depth) };
+        let ret: ExecutionResult = ret;
```

Comment:
So in this case we could do the output type validation here and only return `Option<TypedValue>`.

So the final signature would be `Result<Option<TypedValue>, Error>`.